core/src/test/java/org/elasticsearch/common/util/CountedBitSetTests.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.util;
+
+import org.apache.lucene.util.FixedBitSet;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class CountedBitSetTests extends ESTestCase {
+
+    public void testCompareToFixedBitset() {
+        int numBits = (short) randomIntBetween(8, 4096);
+        final FixedBitSet fixedBitSet = new FixedBitSet(numBits);
+        final CountedBitSet countedBitSet = new CountedBitSet((short) numBits);
+
+        final int iterations = iterations(1000, 20000);
+        for (int i = 0; i < iterations; i++) {
```

Comment:
this is a great tests but I think we can simplify it by randomly setting keys and check at the end that the result is the same for all position (it will also show nothing change for keys after they were set).

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/27547/commits/0e6a31ef894a1a64595c144e1172ae91517138a3